compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1363,6 +1363,29 @@ impl<'a> Parser<'a> {
         })
     }
 
+    /// Parses a field identifier. Specialized version of `parse_ident_common`
+    /// for better diagnostics and suggestions.
+    fn parse_field_ident(&mut self, lo: Span) -> PResult<'a, Ident> {
+        let (ident, is_raw) = self.ident_or_err()?;
+        if !is_raw && ident.is_reserved() {
+            let err = if self.check_fn_front_matter() {
+                let _ = self.parse_fn(&mut Vec::new(), |_| true, lo);
+                let mut err = self.struct_span_err(
+                    lo.to(self.prev_token.span),
+                    "functions are not allowed in struct definitions",
+                );
+                err.help("unlike in C++, Java, and C#, functions are declared in `impl` blocks");
```

Comment:
I think we should say:

```suggestion
                err.help("unlike in C++, Java, and C#, methods are declared in `impl` blocks");
```

instead because you can declare functions outside of `impl` blocks. Specifically, you declare *methods* in `impl` blocks.

Reply:
You can also have associated `fn`s with no receiver, equivalent to Java's static methods, which we don't call methods :)

Reply:
True, but we're saying that methods are declared in `impl` blocks, not that all functions in `impl` blocks are methods ;)

Reply:
Should I write "methods and associated functions", or would that be confusing to a newcomer? Would it be ok to just write "methods"?

Reply:
I think just "methods" is fine since that's probably what the user intended.

Reply:
I think `functions` here are fine.